web-common/src/main/java/edu/unc/lib/boxc/web/common/services/AccessCopiesService.java

```diff
@@ -97,6 +99,11 @@ public boolean hasViewableFiles(ContentObjectRecord briefObj, AccessGroupSet pri
         return resp.getResults().getNumFound() > 0;
     }
 
+    public boolean hasViewablePdf(ContentObjectRecord workObj, AccessGroupSet principals) {
```

Comment:
This method accepts work and file objects, so i'd suggest renaming `workObj` to `contentObj`. Also, could you add javadoc to this method to describe what it does?

You've moved most of the logic out of this method up into the controller, which unfortunately means its not being tested. Up to you if you want to change it or not, but this method could be renamed `getViewablePdfFilePid`, and the method could perform lines 236-255 from the controller, returning the ID of the file object to use in the viewer or null.

Reply:
Yeah, I think a lot of it may need to be moved back over to the service for checking which file to download.